test/node-http-test.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: 'off' */
+
+'use strict';
+
+const assert = require('bsert');
+const Network = require('../lib/protocol/network');
+const FullNode = require('../lib/node/fullnode');
+const rules = require('../lib/covenants/rules');
+const network = Network.get('regtest');
+
+const { NodeClient, WalletClient } = require('hs-client');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+describe('Node http', function() {
+  this.timeout(5000);
+  let NAME0;
+  let node;
+  let miner;
+  let chain;
+  let NAME1;
+
+  const mineBlocks = async (n = 1) => {
+    for (let i = 0; i < n; i++) {
+      const block = await miner.mineBlock();
+      await chain.add(block);
+    }
+  };
+
+  beforeEach(async () => {
+    node = new FullNode({
+      memory: true,
+      apiKey: 'foo',
+      network: 'regtest',
+      workers: true,
+      plugins: [require('../lib/wallet/plugin')]
+    });
+    miner = node.miner;
+    chain = node.chain;
+    NAME0 = await rules.grindName(10, 0, network);
+    NAME1 = await rules.grindName(10, 20, network);
+    await node.open();
+    await mineBlocks(network.names.auctionStart);
+    assert.equal(network.names.auctionStart, 0);
+    await mineBlocks(1);
+  });
+
+  afterEach(async () => {
+    await node.close();
+  });
+
+  describe('getNameInfo', () => {
+    describe('For names that are available at height 0', () => {
+      it('It should return null when there hasn\'t been an auction initiated', async () => {
+        const nameInfo = await nclient.getNameInfo(NAME0);
+        assert.deepEqual(nameInfo, {
+          info: null,
+            start: {
+              reserved: false,
+              start: 0,
+              week: 0
+            }
+        });
+      });
+      it('It should start an auction on the first day', async () => {
+        await mineBlocks(1);
+        const nameInfo = await nclient.getNameInfo(NAME0);
+        assert.deepEqual(nameInfo, {
+          info: null,
+            start: {
+              reserved: false,
+              start: 0,
+              week: 0
+            }
+        });
+        const open = await wclient.execute('sendopen', [NAME0]);
+        assert(open);
+      });
+      it('It should start an auction on the 2nd day', async () => {
+        // Question: This test passes non-deterministically. Why?
```

Comment:
There is a race condition with mining blocks, a hacky way to fix the problem is using a `sleep` function, the better way to do it is to listen to events and `await` for the event you are listening for, otherwise timeout after some period

Reply:
It could be the overhead of the client interacting with the server, but I think it could also be a race condition between the node and the wallet. The older versions of bcoin's wallet shared the same locks with the node, when we decided to separate everything out, we got more modularity, but we ended up in race condition hell. :(

Like @tynes said, the best thing to do is probably wait a couple hundred ms unless someone wants to write some utils that await wallet/node events.

The other possibility is to switch back to having the wallet share the same locks with the node which might mitigate some of this. Something I've wanted for a while.